lib/providers/indoor_data.dart

```diff
@@ -0,0 +1,38 @@
+import 'package:concordia_navigation/models/university.dart';
+import 'package:concordia_navigation/services/dijkstra.dart';
+import 'package:concordia_navigation/services/outdoor/indoor_itinerary.dart';
+import 'package:flutter/material.dart';
+
+class IndoorData extends ChangeNotifier {
+  IndoorItinerary _indoorItinerary;
+
+  IndoorData();
+
+  bool wheelchair = false;
+
+  void toggleWheelchair() {
+    wheelchair = !wheelchair;
+    if (_indoorItinerary != null) {
+      University.loadJson().then((data) {
+        Dijkstra.shortest = Dijkstra.fromJson(data);
+        setItinerary(
+            start: _indoorItinerary.path.first.name,
+            end: _indoorItinerary.path.last.name,
+            accessible: wheelchair);
+      });
```

Comment:
Isn't University.loadJson() and Dijkstra.fromJson(data) already being invoked in SplashScreen?
Why do it here?

Reply:
There was a bug with indoor itineraries where only the first indoor itinerary was correct and any further itinerary was incorrect.